sdks/python/apache_beam/runners/interactive/display/pipeline_graph.py

```diff
@@ -107,6 +135,17 @@ def _top_level_transforms(self):
         top_level_transform_proto = transforms[top_level_transform_id]
         yield top_level_transform_id, top_level_transform_proto
 
+  def _decorate(self, value):
+    """Decorates label-ish values used for rendering in dot language.
+
+    Escapes special characters in the given str value for dot language. Please
+    escape all PTransform unique names when building dot representation.
+    Otherwise, special characters will break the graph rendered.
```

Comment:
Is this statement still valid? Does 'value' need to be escaped any more?

Reply:
Yes, I think the `escape` has a different meaning in DOT's doc. When we say escape (or 
 URL encoding), we mean it automatically adds things like `\` before special characters or convert ` ` into `%20`. 

>In computing and telecommunication, an escape character is a character which invokes an alternative interpretation on subsequent characters in a character sequence. An escape character is a particular case of metacharacters. Generally, the judgment of whether something is an escape character or not depends on context.

But in DOT, it kind of means the opposite.

Because characters like `:` or `[]` have its own meanings in DOT, the recommended way is to enclose them with `"` so that they will be recognized as string literals.
That's why originally the comment says `"` is the only character that cannot be escaped because `"` is the character that escapes everything in-between.
Now, we do a replacement to convert `"` into `\\"` so that it becomes `\"` in dot and rendered as `"`.
However, `\` is still not escaped because it's used as an escape character even in-between `"`. This never ends as long as python literals and DOT work like this.

Reply:
We escape both `\` and `"` and have documented how the process works behind the scene.